pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherSingleActiveConsumer.java

```diff
@@ -520,6 +521,16 @@ private synchronized void internalReadEntriesFailed(ManagedLedgerException excep
                 log.debug("[{}] Error reading transaction entries : {}, - Retrying to read in {} seconds", name,
                         exception.getMessage(), waitTimeMillis / 1000.0);
             }
+        } else if (exception instanceof ManagedLedgerException.CursorAlreadyClosedException) {
+            // Read entry failed because cursor is NonDurableCursor and is closed.
+            // In this case, there is no need to trigger more read operations
+            // since the next read operation would directly skip and finish.
+            // Just wait for a new NonDurableCursor open and read again.
+            if (cursor instanceof NonDurableCursorImpl) {
+                log.debug("[{}-{}] Error reading entries at {} : {} - Retrying to read in {} seconds", name, c,
+                        cursor.getReadPosition(), exception.getMessage(), waitTimeMillis / 1000.0);
```

Comment:
Is this debug log message correct? I don't think that there's a retry in this case.

Reply:
have changed the log message